integration_tests/steps/two_sided_market.py

```diff
@@ -199,3 +200,63 @@ def energy_origin(context, producer, consumer):
                 assert trade.seller_origin == producer
 
     assert trade_count > 0
+
+
+@then('trades are matched only on the Grid market')
+def trades_matched_on_grid(context):
+    # Assert that all grid trades contain Offer objects, bid trades are not tracked
+    assert all(type(trade.offer) == Offer
+               for market in context.simulation.area.past_markets
```

Comment:
I think you need to explicitly set `KEEP_PAST_MARKETS=True` in the setup file being used else it will only be validated for only one last past markets.

Reply:
I was under the impression that KEEP_PAST_MARKETS is True by default for the integration tests, however I will explicitly declare it in the setup file in order to cover our base.